internal/slicer/slicer_test.go

```diff
@@ -769,6 +768,63 @@ var slicerTests = []slicerTest{{
 	report: map[string]string{
 		"/dir/nested/file": "file 0644 84237a05 {test-package_myslice}",
 	},
+}, {
+	summary: "Generate (manifest) directory is not extracted from package if exists",
+	slices:  []setup.SliceKey{{"test-package", "myslice"}},
+	release: map[string]string{
+		"slices/mydir/test-package.yaml": `
+			package: test-package
+			slices:
+				myslice:
+					contents:
+						/dir/nested/**: {generate: manifest}
+		`,
+	},
+	filesystem: map[string]string{},
+	report:     map[string]string{},
+}, {
+	summary: "Multiple slices of same package",
+	slices: []setup.SliceKey{
+		{"test-package", "myslice"}, {"test-package", "manifest"}, {"test-package", "a-slice"},
+	},
+	release: map[string]string{
+		"slices/mydir/test-package.yaml": `
+			package: test-package
+			slices:
+				myslice:
+					contents:
+						/dir/file:
+						/dir/file-copy:  {copy: /dir/file}
+						/other-dir/file: {symlink: ../dir/file}
+						/dir/text-file:  {text: data1}
```

Comment:
Can you add this file in `a-slice` as well? That we we test the inner for loop in your changes.

Reply:
Added in 3aac495dd2894cf79a6eb8dfe3c7422ef619d1c2.